test/org/sagebionetworks/bridge/services/ConsentPdfTest.java

```diff
@@ -125,13 +123,60 @@ public void legacyDocWithInvalidSig() throws Exception {
     public void newDocWithInvalidSig() throws Exception {
         ConsentSignature sig = makeInvalidSignature();
         
-        ConsentPdf consentPdf = new ConsentPdf(study, PST, participant.getEmail(), sig,
-                SharingScope.NO_SHARING, NEW_DOCUMENT_FRAGMENT, consentBodyTemplate);
+        ConsentPdf consentPdf = new ConsentPdf(study, EMAIL_PARTICIPANT, sig, SharingScope.NO_SHARING,
+                NEW_DOCUMENT_FRAGMENT, consentBodyTemplate);
         
         String output = consentPdf.getFormattedConsentDocument();
         validateNewDocBody(output);
     }
+    
+    @Test
+    public void phoneSignature() throws Exception {
+        ConsentSignature sig = makeSignatureWithoutImage();
+        
+        StudyParticipant phoneParticipant = new StudyParticipant.Builder().withPhone(TestConstants.PHONE).withPhoneVerified(true).build();
+        
+        ConsentPdf consentPdf = new ConsentPdf(study, phoneParticipant, sig, SharingScope.NO_SHARING,
+                NEW_DOCUMENT_FRAGMENT, consentBodyTemplate);
+        String output = consentPdf.getFormattedConsentDocument();
+        assertTrue(output.contains(">(971) 248-6796<"));
+        assertTrue(output.contains(">Phone Number<"));
+    }
+    
+    @Test
+    public void externalIdSignature() throws Exception {
+        ConsentSignature sig = makeSignatureWithoutImage();
+        
+        StudyParticipant extIdParticipant = new StudyParticipant.Builder().withExternalId("anId").build();
+        
+        ConsentPdf consentPdf = new ConsentPdf(study, extIdParticipant, sig, SharingScope.NO_SHARING,
+                NEW_DOCUMENT_FRAGMENT, consentBodyTemplate);
+        String output = consentPdf.getFormattedConsentDocument();
+        assertTrue(output.contains(">anId<"));
+        assertTrue(output.contains(">ID<"));
+    }
 
+    @Test(expected = BridgeServiceException.class)
+    public void noVerifiedCommunicationChannelThrows() throws Exception {
+        ConsentSignature sig = makeSignatureWithoutImage();
+        
+        StudyParticipant noChannelParticipant = new StudyParticipant.Builder().build();
+        
+        ConsentPdf consentPdf = new ConsentPdf(study, noChannelParticipant, sig, SharingScope.NO_SHARING,
+                NEW_DOCUMENT_FRAGMENT, consentBodyTemplate);
+        consentPdf.getFormattedConsentDocument();
+    }
+    
+    public void dateFormattedCorrectly() throws Exception {
+        ConsentSignature sig = makeSignatureWithoutImage();
+        
+        ConsentPdf consentPdf = new ConsentPdf(study, EMAIL_PARTICIPANT, sig, SharingScope.NO_SHARING,
+                NEW_DOCUMENT_FRAGMENT, consentBodyTemplate);
+        String output = consentPdf.getFormattedConsentDocument();
+        
+        assertTrue("Contains formatted date", output.contains(ConsentPdf.FORMATTER.print(DateTime.now()) + " (GMT)"));
```

Comment:
It would be better to use DateUtils to mock now and verify the output against a known string. Otherwise, if ConsentPdf.FORMATTER formats the date incorrectly, this test will never catch it.

Reply:
Changed to use a fixed point in time